App.js

```diff
@@ -143,10 +122,10 @@ export default class App extends Component {
         PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION
       );
       if (granted === PermissionsAndroid.RESULTS.GRANTED) {
-        const sdkInitialized = await this.isSdkInitialized();
-        if (sdkInitialized) {
+        const sdkNotInitialized = await this.isSdkNotInitialized();
+        if (!sdkNotInitialized) {
```

Comment:
The double negative is confusing here. Additionally, this condition shouldn't pass if the SDK is still initializing. I suggest adding a second method called `isSdkInitialized` that explicitly checks for the `INITIALIZED` state.